03-rnaseq/pathway-analysis_rnaseq_03_gsva.Rmd

```diff
@@ -0,0 +1,664 @@
+
+Then, we need to filter out the genes that have not been expressed or that have low expression counts. 
+Removing low count genes can also help improve results. 
+We are going to do some pre-filtering to keep only genes with 50 or more reads in total across the samples.
+
+```{r}
+# The next DESeq2 functions need the values to be converted to integers
+expression_df <- round(expression_df) %>%
+  # The next steps require a data frame and round() returns a matrix
+  as.data.frame() %>%
+  # Only keep rows that have total counts above the cutoff
+  dplyr::filter(rowSums(.) >= 50)
+```
+
+## Create a DESeqDataset
+
+We will be using the `DESeq2` package for [normalizing and transforming our data](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/00-intro-to-rnaseq.html#deseq2-transformation-methods), which requires us to format our data into a `DESeqDataSet` object.
+We turn the data frame (or matrix) into a [`DESeqDataSet` object](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/00-intro-to-rnaseq.html#02_About_DESeq2) and specify which variable labels our experimental groups using the [`design` argument](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs) [@Love2014].
+In this chunk of code, we will not provide a specific model to the `design` argument because we are not performing a differential expression analysis.
+
+```{r}
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = expression_df, # Our prepped data frame with counts
+  colData = metadata, # Data frame with annotation for our samples
+  design = ~1 # Here we are not specifying a model
+)
+```
+
+## Perform DESeq2 normalization and transformation
+
+We often suggest normalizing and transforming your data for various applications including for GSVA. 
+We are going to use the `vst()` function from the `DESeq2` package to normalize and transform the data.
+For more information about these transformation methods, [see here](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/00-intro-to-rnaseq.html#deseq2-transformation-methods). 
+
+```{r}
+# Normalize and transform the data in the `DESeqDataSet` object using the `vst()`
+# function from the `DESEq2` R package
+dds_norm <- vst(dds)
+```
+
+At this point, if your data set has any outlier samples, you should look into removing them as they can affect your results. 
+For this example data set, we will skip this step (there are no obvious outliers) and proceed.
+
+Extract the normalized counts to a matrix and make it into a data frame so we can use with tidyverse functions later. 
+
+```{r}
+# Retrieve the normalized data from the `DESeqDataSet`
+normalized_counts <- assay(dds_norm) %>%
+  as.data.frame() %>% # Make into a data frame
+  tibble::rownames_to_column("ensembl_id") # Make Gene IDs into their own column
+```
+
+**IDENTICAL TO MICROARRAY GSVA**
+
+### Import Gene Sets
```

Comment:
I have an idea for _variety_ that I'm not sure I'm super attached to but I'll put it out there - what if we went from `msigdbr` data frame to list format required fro GSVA? That would be a use/series of steps that was more generally applicable to organisms besides human.

Reply:
Worth a try!

Reply:
I don't hate it. See what you think: https://github.com/AlexsLemonade/refinebio-examples/pull/396/commits/9bafff4f44c3d70543a85db38548fc25be508fd8

Reply:
I think it's good (although I will probably have thoughts about specific wording). Didn't add too many lines!